src/core/session.ts

```diff
@@ -424,6 +424,18 @@ export class Session
     }
   }
 
+  sanitizeVisitOptionsForTransfer(options: Partial<VisitOptions>): TransferableVisitOptions {
+    const sanitized: { [key: string]: StructuredCloneValue } = {}
+    Object.entries(options).forEach(([key, value]) => {
+      try {
+        sanitized[key] = structuredClone(value)
+      } catch (_) {
+        // Ignore non-transferable values
+      }
+    })
+    return sanitized
+  }
```

Comment:
> I think that would be nice because it makes it clear that those options are going to be lost when handed to the adapter.

Love it. Done in https://github.com/hotwired/turbo/pull/733/commits/495973c33bbeb34327328756bf3e2708320ed651